File: app/src/main/java/id/coedotz/appbiodata/TampilData.java
package id.coedotz.appbiodata;

import android.app.ProgressDialog;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.util.Log;

import id.coedotz.appbiodata.adapter.AdapterData;
import id.coedotz.appbiodata.api.ApiRequestBiodata;
import id.coedotz.appbiodata.api.Retroserver;
import id.coedotz.appbiodata.model.DataModel;
import id.coedotz.appbiodata.model.ResponsModel;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class TampilData extends AppCompatActivity {
    private RecyclerView mRecycler;
    private RecyclerView.Adapter mAdapter;
    private RecyclerView.LayoutManager mManager;
    private List<DataModel> mItems = new ArrayList<>();
    ProgressDialog pd;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tampil_data);

        pd = new ProgressDialog(this);
        mRecycler = (RecyclerView) findViewById(R.id.recyclerTemp);
        mManager = new LinearLayoutManager(this,LinearLayoutManager.VERTICAL, false);
        mRecycler.setLayoutManager(mManager);

        pd.setMessage("Loading ...");
        pd.setCancelable(false);
        pd.show();

        ApiRequestBiodata api = Retroserver.getClient().create(ApiRequestBiodata.class);
        Call<ResponsModel> getdata = api.getBiodata();
        getdata.enqueue(new Callback<ResponsModel>() {
            @Override
            public void onResponse(Call<ResponsModel> call, Response<ResponsModel> response) {
                pd.hide();
                Log.d("RETRO", "RESPONSE : " + response.body().getKode());
                mItems = response.body().getResult();

                mAdapter = new AdapterData(TampilData.this,mItems);
                mRecycler.setAdapter(mAdapter);
                mAdapter.notifyDataSetChanged();

            }

            @Override
            public void onFailure(Call<ResponsModel> call, Throwable t) {
                pd.hide();
                Log.d("RETRO", "FAILED : respon gagal");

            }
        });

    }
}

File: app/src/main/java/id/coedotz/appbiodata/TampilData.kt
package id.coedotz.appbiodata

import android.app.ProgressDialog
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import id.coedotz.appbiodata.adapter.AdapterData
import id.coedotz.appbiodata.api.ApiRequestBiodata
import id.coedotz.appbiodata.api.Retroserver
import id.coedotz.appbiodata.model.DataModel
import id.coedotz.appbiodata.model.ResponsModel
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class TampilData : AppCompatActivity() {
    private lateinit var mRecycler: RecyclerView
    private lateinit var mAdapter: RecyclerView.Adapter<*>
    private lateinit var mManager: RecyclerView.LayoutManager
    private var mItems: MutableList<DataModel> = ArrayList()
    private lateinit var pd: ProgressDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tampil_data)

        pd = ProgressDialog(this)
        mRecycler = findViewById(R.id.recyclerTemp)
        mManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        mRecycler.layoutManager = mManager

        pd.setMessage("Loading ...")
        pd.setCancelable(false)
        pd.show()

        val api = Retroserver.getClient().create(ApiRequestBiodata::class.java)
        val getdata: Call<ResponsModel> = api.getBiodata()
        getdata.enqueue(object : Callback<ResponsModel> {
            override fun onResponse(call: Call<ResponsModel>, response: Response<ResponsModel>) {
                pd.hide()
                Log.d("RETRO", "RESPONSE : " + response.body()?.kode)
                mItems = (response.body()?.result ?: ArrayList()) as MutableList<DataModel>

                mAdapter = AdapterData(this@TampilData, mItems)
                mRecycler.adapter = mAdapter
                mAdapter.notifyDataSetChanged()
            }

            override fun onFailure(call: Call<ResponsModel>, t: Throwable) {
                pd.hide()
                Log.d("RETRO", "FAILED : respon gagal")
            }
        })
    }
}